src/mgwasm.c

```diff
@@ -0,0 +1,75 @@
+#include "mgwasm.h"
+
+#ifdef __EMSCRIPTEN__
+
+#include <stddef.h>
+#include <stdio.h>
+#include <sys/select.h>
+
+int read_loop(const int sock) {
+  fd_set fdr;
+  FD_ZERO(&fdr);
+  FD_SET(sock, &fdr);
+  int poll = select(sock + 1, &fdr, NULL, NULL, NULL);
+  if (poll == -1) {
+    return -1;
+  }
+  if (!FD_ISSET(sock, &fdr)) {
+    return -100;
+  }
+  return 1;
+}
+
+int write_loop(const int sock) {
+  fd_set fdw;
+  FD_ZERO(&fdw);
+  FD_SET(sock, &fdw);
+  int poll = select(sock + 1, NULL, &fdw, NULL, NULL);
+  if (poll == -1) {
+    return -1;
+  }
+  if (!FD_ISSET(sock, &fdw)) {
+    return -100;
+  }
+  return 1;
+}
+
+static const size_t DELAY_MS = 10;
+
+int mg_yield_until_async_read_sock(const int sock) {
+  while (1) {
+    emscripten_sleep(DELAY_MS);
+    int res = read_loop(sock);
+    if (res == 1) {
+      return 1;
+    }
+    if (res == -1) {
+      return -1;
+    }
+  }
+}
+
+int mg_yield_until_async_write_sock(const int sock) {
+  while (1) {
+    emscripten_sleep(DELAY_MS);
+    int res = write_loop(sock);
+    if (res == 1) {
+      return 1;
+    }
+    if (res == -1) {
+      return -1;
+    }
+  }
+}
+
+int mg_yield_until_async_read(const mg_transport *transport) {
+  int sock = ((mg_raw_transport *)transport)->sockfd;
+  return mg_yield_until_async_read_sock(sock);
+}
+
+int mg_yield_until_async_write(const mg_transport *transport) {
+  int sock = ((mg_raw_transport *)transport)->sockfd;
+  return mg_yield_until_async_write_sock(sock);
+}
```

Comment:
I think we need another layer of abstraction here, as `mg_transport` can be something else, that is not `mg_raw_transport`. Currently it is not possible, because the secure transport is ifdef-ed out, but this is error prone. I would add another function pointer to `mg_transport` that can be null (that case this function doesn't call anything) or can point to `mg_raw_until_async_read` in case of emscripten.